### PR TITLE
Reduce shared volumes to a minimum using initContainers, secrets and other Kubernetes primitives

### PR DESCRIPTION
**Pull request by palkra**
### Reasons for this PR:
 - The network is based on tutorials that utilize shared storage between all containers which makes our network a monolith.
 - We should utilize more of Kubernetes features to encapsulate more logic within the pods.
 - We use a hostPath mount to share all files, in theory only some files need to be shared between containers. Switching to a distributed Kubernetes cluster is not possible at this point.

### Changes in this PR:
 - We generate our own root certificates for each CA using OpenSSL and store them in Kubernetes secrets. The certs are provided to other pods using the secrets instead of copying them.
 - We attached enrollments to the pod that uses the keys of the enrolled user as an initContainer instead of having separated jobs with shared volumes.
 - We replaced hostPath mounts by emptyDirs sharing files only within the Pod instead of globally.
 - We use the fabric-tools container for generating the genesis block as an initContainer to the orderer instead of generating it in the deploy script.
 - We got rid of templating the configtx since paths on the initContainer for the genesis block are fixed as opposed to hostPaths used before. The configtx is mounted to the container using a configMap instead of using a hostPath mount.
 - There are only three shared volumes remaining, one for providing the scripts to all pods, one for providing the chaincode, and one containing public keys and public channel files that we could not store in secrets. These could further be reduced to only one shared storage with only a few public files, which allows us to deploy the network to distributed Kubernetes clusters using for example Google Cloud Firestore.

### CODE REVIEW

File: scripts/startNetwork/registerUsers/registerOrdererOrgUsers.sh [hunk 1]
```diff
@@ -11,7 +11,6 @@ export CA_ORDERER_HOST=0.0.0.0:7053
 export FABRIC_CA_CLIENT_TLS_CERTFILES=ca-cert.pem
 export FABRIC_CA_CLIENT_HOME=/tmp/hyperledger/ca-client/
 mkdir -p $FABRIC_CA_CLIENT_HOME
-cp /tmp/hyperledger/fabric-ca/crypto/ca-cert.pem $FABRIC_CA_CLIENT_HOME/$FABRIC_CA_CLIENT_TLS_CERTFILES
```
Comment (palkra) [inline scripts/startNetwork/registerUsers/registerOrdererOrgUsers.sh:14]: Something is missing here

File: scripts/startNetwork/registerUsers/registerTLSusers.sh [hunk 2]
```diff
@@ -7,10 +7,9 @@ set -e
 log "Use CA-client to enroll admin"
 
 export TLS_CA_HOST=0.0.0.0:7052
-export FABRIC_CA_CLIENT_TLS_CERTFILES=tls-ca-cert.pem
+export FABRIC_CA_CLIENT_TLS_CERTFILES=/tmp/secrets/cert.pem
```
Comment (palkra) [inline scripts/startNetwork/registerUsers/registerTLSusers.sh:10]: Path needs to be relative to  FABRIC_CA_CLIENT_HOME

**Comment by palkra** [comment]
@NikoBergemann we need some help to identify the certificates we need to provide for the API to work.

**Comment by NikoBergemann** [comment]
> @NikoBergemann we need some help to identify the certificates we need to provide for the API to work.

just need the tls-ca-cert and the org1-ca-cart

File: scripts/startNetwork.sh [hunk 3]
```diff
@@ -23,39 +24,18 @@ source ./scripts/startNetwork/fixPrepareHostPath.sh
 small_sep
 kubectl create -f k8s/namespace.yaml
 
+faketime -m -f -1d /bin/bash -c scripts/startNetwork/generateSecrets.sh
```
Comment (palkra) [inline scripts/startNetwork.sh:27]: Open question: How to ensure that certificates are valid directly after creation even if the clock of Kubernetes is slightly drifted.

**Review by anemonekk — APPROVED** [review]

**Pull request merged**